libraries/shared/watcher/storage_watcher.go

```diff
@@ -58,14 +67,33 @@ func (storageWatcher StorageWatcher) AddTransformers(initializers []transformer.
 	}
 }
 
-func (storageWatcher StorageWatcher) Execute(diffsChan chan utils.StorageDiff, errsChan chan error, queueRecheckInterval time.Duration) {
+// BackFill configures the StorageWatcher to backfill missed storage diffs using a modified archival geth client
+func (storageWatcher StorageWatcher) BackFill(backFiller storage.BackFiller, minDeploymentBlock int) {
+	// need to learn which block to perform the backfill process up to
+	// this blocks until the Execute process sends us the first block number it sees
+	startingSyncBlock := <-storageWatcher.StartingSyncBlockChan
+	backFilledDiffs, err := backFiller.BackFill(uint64(minDeploymentBlock), uint64(startingSyncBlock))
+	if err != nil {
+		storageWatcher.ErrsChan <- err
+	}
+	for _, storageDiff := range backFilledDiffs {
+		storageWatcher.DiffsChan <- storageDiff
+	}
+}
+
+func (storageWatcher StorageWatcher) Execute(queueRecheckInterval time.Duration, backFill bool) {
 	ticker := time.NewTicker(queueRecheckInterval)
-	go storageWatcher.StorageFetcher.FetchStorageDiffs(diffsChan, errsChan)
+	go storageWatcher.StorageFetcher.FetchStorageDiffs(storageWatcher.DiffsChan, storageWatcher.ErrsChan)
+	start := true
 	for {
 		select {
-		case fetchErr := <-errsChan:
+		case fetchErr := <-storageWatcher.ErrsChan:
 			logrus.Warn(fmt.Sprintf("error fetching storage diffs: %s", fetchErr))
-		case diff := <-diffsChan:
+		case diff := <-storageWatcher.DiffsChan:
+			if start && backFill {
+				storageWatcher.StartingSyncBlockChan <- diff.BlockHeight - 1
+				start = false
```

Comment:
I think it's saying that if we're fetching storage diffs, and we're back filling then we should pass the current block -1 to this channel which is then used by the backfiller to know where to backfill until, is that right?

Based on that assumption, a couple follow up questions: 
1. why block height **-1**?
2. If we decide that we no longer care about the order that diffs come in (based on our IPM discussion earlier today), this piece shouldn't be affected, since we'd still want to stop backfilling at the block that the subscription has started. Is that right?

Reply:
1. `blockheight - 1` because since we received the diff at `blockheight` over the subscription we only need to backfill up til the block height before it.
2. That's right, the range we define for doing the backfill over will not be affected by the changes to the processing order.